### PR TITLE
Update WR (radial gradients, generate_frame API, mac subpixel improvements, ANGLE shader support).

### PR DESCRIPTION
**Pull request by glennw**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/15111)

### CODE REVIEW

**Comment by Ms2ger** [comment]
```
./Cargo.lock:1: duplicate versions for package `core-graphics`
	The following packages depend on version 0.4.2 from 'crates.io':
		azure
		cocoa
		core-text
		gfx
		servo-glutin
	The following packages depend on version 0.5.0 from 'crates.io':
		webrender
		webrender_traits
```

**Comment by glennw** [comment]
@Ms2ger Ugh - I guess that means this can't land until those other packages are updated to 0.5.0? That's unfortunate timing :|

**Comment by Manishearth** [comment]
r=me when conflcits are resolved

**Comment by jdm** [comment]
All other packages are published. This should be ready to go.

**Comment by Manishearth** [comment]
Ooh radial gradiants, finally 😀

**Comment by jdm** [comment]
```
  ▶ TIMEOUT [expected FAIL] /css21_dev/html4/abspos-containing-block-009.htm
  │ 
  │ VMware, Inc.
  │ Gallium 0.4 on softpipe
  │ 3.3 (Core Profile) Mesa 12.0.1
  └ Shutting down the Constellation after generating an output file or exit flag specified

  ▶ TIMEOUT [expected FAIL] /css21_dev/html4/abspos-containing-block-008.htm
  │ 
  │ VMware, Inc.
  │ Gallium 0.4 on softpipe
  │ 3.3 (Core Profile) Mesa 12.0.1
  └ Shutting down the Constellation after generating an output file or exit flag specified
```

**Comment by glennw** [comment]
I'm unable to reproduce these failures locally. I'll try on some other machines next week.

**Comment by jdm** [comment]
The bidi failure is #15182.

**Comment by glennw** [comment]
The bidi failure *may* have become a permanent failure with this update. I'm investigating now to see if it's related to the WR update itself or a coincidence.

**Comment by glennw** [comment]
The bidi failure is occurring reliably for me with this PR, but as far as I can tell is due to timing differences only, exposing a race condition in layout (the text run gets broken at a different point).

I'm unable to reproduce the other failures locally yet - I'll try to repro these on a different machine.

**Comment by glennw** [comment]
@bors-servo try
 * Updating PR to include latest tiling changes in WR. Let's see if it makes any difference on CI while I investigate the (seemingly unrelated) intermittent failures locally.

**Comment by glennw** [comment]
@bors-servo retry
 * https://github.com/servo/servo/issues/15239 - Let's see if the iframe hide_and_show is intermittent.

**Comment by jdm** [comment]
Multiple people have been asking about hide_and_show.html in the past couple days; it seems like it's only appeared in try runs so nobody has filed and issue about it.

**Comment by glennw** [comment]
@bors-servo try
 * I have disabled the bidi-008b test - the display lists are significantly different between the ref image and the test, so this failure / race condition is unrelated to the WR update PR.

**Comment by glennw** [comment]
I can (sometimes) reproduce those timeouts locally now.

The root cause is that our CI machines are trying to run too many processes at once and some tests are timing out due to our huge thread counts.

However, I am trying out a workaround / patch for WR that should make the software renderer run a lot faster in these set of tests (those with large borders). Hopefully that should help out with reducing the likelihood of these tests timing out on CI.

**Comment by glennw** [comment]
See https://github.com/servo/webrender/pull/793

**Comment by glennw** [comment]
@bors-servo retry
 * Suspect intermittent - let's see.

**Comment by glennw** [comment]
@bors-servo try

Oh! That was a new pass. Updated expectations, let's try again.

**Comment by glennw** [comment]
@bors-servo retry
 * https://github.com/servo/servo/issues/14378

File: components/net/image_cache_thread.rs [hunk 1]
```diff
@@ -33,6 +33,25 @@ use webrender_traits;
 /// MAYBE(Yoric):
 ///     * For faster lookups, it might be useful to store the LoadKey in the DOM once we have performed a first load.
 
+// TODO(gw): This is a port of the old is_image_opaque code from WR.
```
Comment (pcwalton) [inline components/net/image_cache_thread.rs]: Add a comment saying that this is slow and needs SIMD

Comment (glennw) [inline components/net/image_cache_thread.rs]: Fixed

**Pull request merged**